boards/DIM/Src/main.c

```diff
@@ -437,11 +480,35 @@ void RunTaskCanTx(void const *argument)
     /* Infinite loop */
     for (;;)
     {
-        osDelay(1);
+        Io_SharedCan_TransmitEnqueuedCanTxMessagesFromTask();
     }
     /* USER CODE END RunTaskCanTx */
 }
 
+/* USER CODE BEGIN Header_StartTask1kHz */
+/**
+ * @brief Function implementing the Task1kHz thread.
+ * @param argument: Not used
+ * @retval None
+ */
+/* USER CODE END Header_StartTask1kHz */
+void StartTask1kHz(void const *argument)
+{
+    /* USER CODE BEGIN StartTask1kHz */
+    UNUSED(argument);
+    uint32_t                PreviousWakeTime = osKernelSysTick();
+    static const TickType_t period_ms        = 1;
+
+    /* Infinite loop */
+    for (;;)
+    {
+        Io_CanTx_EnqueuePeriodicMsgs(
+            can_tx, osKernelSysTick() * portTICK_PERIOD_MS);
+        osDelayUntil(&PreviousWakeTime, period_ms);
+    }
+    /* USER CODE END StartTask1kHz */
+}
```

Comment:
Please correct me if I'm wrong, but generally we have dedicated tasks for can RX and TX, no? Any particular reason we don't here?

Reply:
We do! 

```
void        RunTask100Hz(void const *argument);
void        RunTaskCanRx(void const *argument);
void        RunTaskCanTx(void const *argument);
void        StartTask1kHz(void const *argument);
```

The task order is just different on this board haha. I can go edit the `.ioc` so the CanRx and CanTx tasks are at the bottom like the other projects.

Reply:
Unless you're asking about something else? :thinking:

Reply:
Nope, sorry, somehow just totally missed them here.